tests/learner/generics_learner_test.py

```diff
@@ -0,0 +1,165 @@
+import random
+from typing import Iterable
+
+import pytest
+
+from adam.axes import AxesInfo
+from adam.curriculum.curriculum_utils import PHASE1_CHOOSER_FACTORY, phase1_instances
+from adam.curriculum.phase1_curriculum import (
+    _make_generic_statements_curriculum,
+    _make_eat_curriculum,
+    _make_drink_curriculum,
+    _make_sit_curriculum,
+    _make_jump_curriculum,
+    _make_fly_curriculum,
+    _make_colour_predicates_curriculum,
+    _make_kind_predicates_curriculum,
+    _make_objects_with_colors_curriculum,
+)
+from adam.language.language_utils import phase1_language_generator
+from adam.learner import LearningExample
+from adam.learner.attributes import SubsetAttributeLearnerNew
+from adam.learner.generics import SimpleGenericsLearner
+from adam.learner.integrated_learner import IntegratedTemplateLearner
+from adam.learner.language_mode import LanguageMode
+from adam.learner.plurals import SubsetPluralLearnerNew
+from adam.learner.verbs import SubsetVerbLearnerNew
+from adam.ontology.ontology import Ontology
+from adam.ontology.phase1_ontology import (
+    GAILA_PHASE_1_ONTOLOGY,
+    PHASE_1_CURRICULUM_OBJECTS,
+    LIQUID,
+    is_recognized_particular,
+)
+from adam.random_utils import RandomChooser
+from adam.situation import SituationObject
+from adam.situation.high_level_semantics_situation import HighLevelSemanticsSituation
+from tests.learner import LANGUAGE_MODE_TO_TEMPLATE_LEARNER_OBJECT_RECOGNIZER
+
+
+def integrated_learner_factory(language_mode: LanguageMode):
+    rng = random.Random()
+    rng.seed(0)
+    return IntegratedTemplateLearner(
+        object_learner=LANGUAGE_MODE_TO_TEMPLATE_LEARNER_OBJECT_RECOGNIZER[language_mode],
+        attribute_learner=SubsetAttributeLearnerNew(
+            ontology=GAILA_PHASE_1_ONTOLOGY, beam_size=5, language_mode=language_mode
+        ),
+        plural_learner=SubsetPluralLearnerNew(
+            ontology=GAILA_PHASE_1_ONTOLOGY, beam_size=5, language_mode=language_mode
+        ),
+        action_learner=SubsetVerbLearnerNew(
+            ontology=GAILA_PHASE_1_ONTOLOGY, beam_size=5, language_mode=language_mode
+        ),
+        generics_learner=SimpleGenericsLearner(),
+    )
+
+
+def run_generics_test(learner, language_mode):
```

Comment:
Doesn't necessarily need fixing, but to confirm, the test here is just that this doesn't throw any errors?

Reply:
Yes, and also the comments contain the outline for generating the output, which we will need for experiments